sdk/eventhub/azure-eventhub/azure/eventhub/_pyamqp/aio/_authentication_async.py

```diff
@@ -4,35 +4,131 @@
 # license information.
 #-------------------------------------------------------------------------
 from functools import partial
+import time
+from typing import Awaitable, Callable, Optional, Tuple, Union, Any
+
+from ..sasl import SASLAnonymousCredential
 
 from ..authentication import (
     _generate_sas_access_token,
-    SASTokenAuth,
-    JWTTokenAuth
+    AccessToken
 )
-from ..constants import AUTH_DEFAULT_EXPIRATION_SECONDS
-
+from ..constants import AUTH_DEFAULT_EXPIRATION_SECONDS, AUTH_TYPE_CBS, TOKEN_TYPE_JWT, TOKEN_TYPE_SASTOKEN
 
-async def _generate_sas_token_async(auth_uri, sas_name, sas_key, expiry_in=AUTH_DEFAULT_EXPIRATION_SECONDS):
+async def _generate_sas_token_async(
+        auth_uri: str,
+        sas_name: str,
+        sas_key: str,
+        expiry_in: float = AUTH_DEFAULT_EXPIRATION_SECONDS
+    ) -> AccessToken:
     return _generate_sas_access_token(auth_uri, sas_name, sas_key, expiry_in=expiry_in)
 
+class _CBSAuthAsync:
```

Comment:
where/why was this added

Reply:
I added this to separate out async and sync auth. Currently our async auth inherit from sync which makes typing hard